ledger/query/src/static_query.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2019-2025 Provable Inc.
+// This file is part of the snarkVM library.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::QueryTrait;
+use console::{network::prelude::*, program::StatePath, types::Field};
+
+use anyhow::{Result, anyhow};
+use serde::Deserialize;
+use std::{collections::HashMap, str::FromStr};
+
+#[derive(Clone)]
+pub struct StaticQuery<N: Network> {
+    block_height: u32,
+    state_root: N::StateRoot,
+    state_paths: HashMap<Field<N>, StatePath<N>>,
+}
+
+impl<N: Network> StaticQuery<N> {
+    pub fn new(block_height: u32, state_root: N::StateRoot, state_paths: HashMap<Field<N>, StatePath<N>>) -> Self {
+        Self { block_height, state_root, state_paths }
+    }
+}
+
+#[derive(Deserialize)]
+struct StaticQueryInput {
+    state_root: String,
+    height: u32,
+}
+
+impl<N: Network> TryFrom<&String> for StaticQuery<N> {
```

Comment:
I'd implement [`FromStr`](https://doc.rust-lang.org/beta/std/str/trait.FromStr.html) here as it implies parsing more so than `TryFrom<&str>` ([see _the_ book](https://doc.rust-lang.org/nightly/book/ch04-03-slices.html#string-slices-as-parameters) on `&String` vs `&str`).